app/src/main/kotlin/com/hippo/ehviewer/spider/SpiderDen.kt

```diff
@@ -195,6 +196,9 @@ class SpiderDen(val info: GalleryInfo) {
             outFile.openOutputStream().use {
                 response.bodyAsChannel().copyTo(it.channel)
             }
+            val expected = FileHashRegex.find(url)?.run { groupValues[0] }
+            val actual = outFile.sha1()
```

Comment:
Ideally we should update sha before write to the file to improve performance.

Reply:
The avarage time for hashing during testing is < 5 ms, so I didn't bother.
I did not test on legacy devices though.